src/main/java/nl/knaw/dans/ttv/DdTransferToVaultApplication.java

```diff
@@ -17,28 +17,60 @@
 package nl.knaw.dans.ttv;
 
 import io.dropwizard.Application;
+import io.dropwizard.db.PooledDataSourceFactory;
+import io.dropwizard.hibernate.HibernateBundle;
 import io.dropwizard.setup.Bootstrap;
 import io.dropwizard.setup.Environment;
+import nl.knaw.dans.ttv.core.Inbox;
+import nl.knaw.dans.ttv.core.TransferItem;
+import nl.knaw.dans.ttv.db.TransferItemDAO;
+import nl.knaw.dans.ttv.tasks.TransferTask;
+
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingDeque;
 
 public class DdTransferToVaultApplication extends Application<DdTransferToVaultConfiguration> {
 
     public static void main(final String[] args) throws Exception {
         new DdTransferToVaultApplication().run(args);
     }
 
+    private final HibernateBundle<DdTransferToVaultConfiguration> hibernateBundle = new HibernateBundle<DdTransferToVaultConfiguration>(TransferItem.class) {
+        @Override
+        public PooledDataSourceFactory getDataSourceFactory(DdTransferToVaultConfiguration ddTransferToVaultConfiguration) {
+            return ddTransferToVaultConfiguration.getDataSourceFactory();
+        }
+    };
+
     @Override
     public String getName() {
         return "Dd Transfer To Vault";
     }
 
     @Override
     public void initialize(final Bootstrap<DdTransferToVaultConfiguration> bootstrap) {
-        // TODO: application initialization
+        bootstrap.addBundle(hibernateBundle);
     }
 
     @Override
     public void run(final DdTransferToVaultConfiguration configuration, final Environment environment) {
+        final TransferItemDAO transferItemDAO = new TransferItemDAO(hibernateBundle.getSessionFactory());
+        final List<Inbox> inboxes = configuration.buildInboxes();
+        final TransferTask transferTask = new TransferTask(inboxes, transferItemDAO);
+
+        /*adds the transfer task to a callable list which can be invoked with: curl -X POST http://localhost:20001/tasks/nl.knaw.dans.ttv.tasks.TransferTask
+        https://www.dropwizard.io/en/latest/manual/core.html#tasks*/
+        environment.admin().addTask(transferTask);
+
+        //TODO make config parameters configurable
```

Comment:
As far as I can see there is no configuration block for this yet, so you will have to create that. I would like maxThreads, minThreads and maxQueue size to be configurable. The backing class for the queue can stay hard-coded to LinkedBlockingDeque.

Reply:
@janvanmansum so it needs new configuration block and I cannot use?:
```
server:
  applicationContextPath: /
  adminContextPath: /
  applicationConnectors:
    - type: http
      # Replace with port number unique for this service
      port: 20000
  adminConnectors:
    - type: http
      # Replace with port number unique for this service
      port: 20001
  maxThreads: 100
  maxQueuedRequests: 100
```
I've been trying to reach those variables, but that hasn't been easy... It might not be what they are intended. I'm not sure...

Reply:
found it!

Reply:
No, I think those are intended for configuring the Jetty server and the thread pool it uses to dispatch requests. We should be able to configure our own worker threads separately, so that is why se should not use this configuration.